### ISSUE TITLE
Make Smile redirection softer; potentially add "remind me later"

### ISSUE THREAD

**Issue by jborichevskiy** [jamieallen59/greencart#2]
I see two potential issues with the current redirection:
1. It's not possible to use Amazon without signing in (as you get stuck in the Smile login page) without disabling the plugin
2. Upon first login for a non-Smile user, it may be jarring to get redirected straight to the Smile sign-up page without context or explanation. 

I think this might be solved by _detecting_ that the user is not on Smile first, and then recommending (via popup?) that they either read more, sign up, or ignore this message for now. It can also say something like "Once you register with Smile, the extension will redirect you to Amazon Smile so you don't forget".

Sign Up (or Enable) -> would redirect to Smile registration
Read More -> Taken to a short blog post that I can write up with a little more context and a few charity suggestions (since there's a few hundred)
Ignore -> Disables the popup until then

Also perhaps "Learn more about Amazon Smile" can be an option in the actual plugin element that would re-trigger the above menu/flow.

Thoughts @jamieallen59 ?

**Comment by jamieallen59** [comment]
Yea I've been thinking about this too. More from the standpoint of how we show that one of our features has been used.

My worry about a popup is that it becomes almost the same feature as what Amazon does itself. We could be more bullish and redirect (to either sign up or directly) and have e.g.

1. Directs to sign up -> small popup near the plugin logo saying 'Why did we do this?' linking to the blog you mentioned (def good idea about recommended charities)
2. Normal redirect -> a popup that instantly fades away just saying e.g. 'We redirected you to Amazon Fresh'

I like the idea of the 'Learn more about Amazon Smile' link too.

I think with this we generally have to come from the assumption (that I have) that everyone always wants to use Smile. Why would I not want some of my money to go to Charity when I use Amazon? Which is why I think as long as we explain it properly then it's ok to be relatively pushy.

I like the idea of the "Once you register with Smile, the extension will redirect you to Amazon Smile so you don't forget" quote too.

To summarise, I think we probably need:
1. A basic non-intrusive notification system to explain when we've done something helpful
2. Add a notification with a link for 'Why did we do this' for non-signed up customers
3. Add a notification that disappears for users who are signed up to smile and get redirected
4. A link within the plugin itself to go to the blog post you mentioned

Number 4 creates the need for a design for what displays when you click the plugin icon which I can do the basics for.

What do you think?

**Comment by jborichevskiy** [comment]
Agreed about a popup being lost in the noise of Amazon's already cluttered UI.

So in that case I'd say the notification is only necessary in the first scenario (`Directs to sign up`) and not 2 (`Normal redirect`) - since if the charity is chosen the redirect isn't even noticeable, which is perfect.

To your summary: definitely makes sense. 

I'd add one thing to that: disable redirection if user is not signed in. This will keep that first loop I mentioned from happening (where the site is redirecting to the Smile sign-in page regardless of link or page viewed). I'm not sure it's worth trying to intercept all sign-in paths since I assume there are quite many *.amazon.com links that people go to without signing in.

Apart from that - your plan makes sense.
